clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerHeartbeatRequestManager.java

```diff
@@ -262,6 +262,9 @@ public ConsumerGroupHeartbeatRequestData buildRequestData() {
             // InstanceId - set if present
             membershipManager.groupInstanceId().ifPresent(data::setInstanceId);
 
+            // RackId - set if present
+            membershipManager.rackId().ifPresent(data::setRackId);
```

Comment:
I think that it should only be set if it did not change since the last request or if we send a full request.

Reply:
Updated it. Thanks.